Formula/subfinder.rb

```diff
@@ -0,0 +1,56 @@
+require "language/go"
+class Subfinder < Formula
+  desc "Subdomain discovery tool"
+  homepage "https://github.com/subfinder"
+  head "https://github.com/subfinder/subfinder.git"
+
+  stable do
+    url "https://github.com/subfinder/subfinder/archive/1.2.tar.gz"
+    sha256 "2b8a09506466baa04c10404d38e0da6a2066a79e8b5c69a720d939804375ba2c"
+    go_resource "github.com/bogdanovich/dns_resolver" do
+      url "https://github.com/bogdanovich/dns_resolver.git",
+            :revision => "a8e42bc6a5b6c9a93be01ca204be7e17f7ba4cd2"
+    end
+
+    go_resource "github.com/subfinder/urlx" do
+      url "https://github.com/subfinder/urlx.git",
+            :revision => "8e731c8be06edbae81cab15937cd3c291c2a7680"
+    end
+    go_resource "github.com/miekg/dns" do
+      url "https://github.com/miekg/dns.git",
+            :revision => "22cda6dc4f8b43ab129d7e79c24975180a0aace6"
+    end
+    go_resource "golang.org/x/crypto" do
+      url "https://go.googlesource.com/crypto.git",
+            :revision => "86a70503ff7e82ffc18c7b0de83db35da4791e6a"
+    end
+    go_resource "golang.org/x/net" do
+      url "https://go.googlesource.com/net.git",
+            :revision => "5ee1b9f4859acd2e99987ef94ec7a58427c53bef"
+    end
+    go_resource "golang.org/x/sys" do
+      url "https://go.googlesource.com/sys.git",
+            :revision => "ce4227a45e2eb77e5c847278dcc6a626742e2945"
+    end
+  end
+
+  depends_on "go" => :build
+
+  def install
+    ENV["GOPATH"] = buildpath
+    ENV["GOBIN"] = bin
+
+    path = buildpath/"src/github.com/subfinder/subfinder"
+    path.install buildpath.children
+
+    Language::Go.stage_deps resources, buildpath/"src" if build.stable?
+    cd path do
+      system "go", "install", "-v", ".../subfinder"
+      prefix.install_metafiles
+    end
+  end
+
+  test do
+    system "#{bin}/subfinder"
```

Comment:
We need a test that exercises the some of the functionality of the app. Version checks or usage checks (foo --version or foo --help) are not sufficient, as explained in the formula cookbook.

In most cases, a good test would involve running a simple test case: run #{bin}/foo input.txt.

- Then you can check that the output is as expected (with assert_equal or assert_match on the output of shell_output)
- You can also check that an output file was created, if that is expected: assert_predicate testpath/"output.txt", :exist?

Some advice for specific cases:

- If the formula is a library, compile and run some simple code that links against it. It could be taken from upstream's documentation / source examples.
- If the formula is for a GUI program, try to find some function that runs as command-line only, like a format conversion, reading or displaying a config file, etc.
- If the software cannot function without credentials, a test could be to try to connect with invalid credentials (or without credentials) and confirm that it fails as expected.
- Same if the software requires a virtual machine, docker instance, etc. to be running.

Reply:
It's a bit tricky testing for functionality since all of the functionality requires connecting to websites which would make some flaky tests. I checked to see if the configuration file gets created.

Reply:
You could check brew.sh. Might alert homebrew of infrastructure issues if it fails.

Reply:
Is the test I added okay?

Reply:
The current test doesn't seem to check any features,  it just runs the executable

Reply:
Whoops didnt push correctly, I just am checking to see if a file is created like the guide example